draft-ietf-httpbis-http2bis.xml

```diff
@@ -3054,9 +3054,18 @@
             </t>
             <t>
               For malformed requests, a server MAY send an HTTP response prior to closing or
-              resetting the stream.  Clients MUST NOT accept a malformed response. Note that these
-              requirements are intended to protect against several types of common attacks against
-              HTTP; they are deliberately strict because being permissive can expose
+              resetting the stream.  Clients MUST NOT accept a malformed response.
+            </t>
+            <t>
+              Endpoints that progressively process messages might have performed some processing
+              before detecting any error.  For instance, it might be possible to generate an
+              informational or 404 status code without having received a complete request.  A
+              malformed request or response that is partially or completely processed before being
+              identified as malformed MUST NOT be treated as having completed successfully.
```

Comment:
Hmm, now that I reread this I feel a bit conflicted by this. MUST NOT be treated by whom?

Reply:
Oh yeah, that really isn't clear at all.

Maybe what this boils down to is that you can't use the REFUSED_STREAM if you do any processing of partial messages.

I'll give this another go.